app/presenters/tree_node/menu/item.rb

```diff
@@ -0,0 +1,36 @@
+module TreeNode
+  module Menu
+    class Item < TreeNode::Menu::Node
+      set_attribute(:key)      { "#{@options[:node_id_prefix]}__#{@object.feature}" }
+      set_attribute(:title)    { _(details[:name]) }
+      set_attribute(:tooltip)  { _(details[:description]) || _(details[:name]) }
+      set_attribute(:selected) { self_selected? || select_by_children_selected }
+
+      def children
```

Comment:
It doesn't _generate_ (build the node for) it's own children, that's done in the `x_get_tree_*_kids` methods, but the states for child nodes are _counted_ here to correctly determine the select state.